contracts/ExampleNFT.cdc

```diff
@@ -12,18 +13,67 @@ pub contract ExampleNFT: NonFungibleToken {
     pub event Withdraw(id: UInt64, from: Address?)
     pub event Deposit(id: UInt64, to: Address?)
 
-    pub resource NFT: NonFungibleToken.INFT {
+    pub let CollectionStoragePath: StoragePath
+    pub let CollectionPublicPath: PublicPath
+    pub let MinterStoragePath: StoragePath
+
+    pub resource NFT: NonFungibleToken.INFT, MetadataViews.Resolver {
         pub let id: UInt64
 
-        pub var metadata: {String: String}
+        pub let name: String
+        pub let thumbnail: String
+        pub let description: String
+
+        init(
+            id: UInt64,
+            name: String,
+            description: String,
+            thumbnail: String,
+        ) {
+            self.id = id
+            self.name = name
+            self.thumbnail = thumbnail
+            self.description = description
+        }
+    
+        pub fun getViews(): [Type] {
+            return [
```

Comment:
I think this should add String that just resolves to self.name
I also think it should add Idenitty.

Reply:
What is Identity? Also, is there a specific definition for what name would mean across different NFT projects or is it completely up to the developer?

Reply:
In my eyes name is up to the developer. 

My Identity suggestion from up above is because I think exposing UUID is something that should really be done on all resources. It is so much more convenient to work with a UUID compared to the type and the ID to uniquely identify a resource. The reason i added Type and ID there was in case somebody needs to do a mapping.